src/containerregistry/docker_keychain.go

```diff
@@ -0,0 +1,82 @@
+package containerregistry
```

Comment:
May be I missed something, but it's not clear to me why this package has the name `containerregistry` ?

For example:
>Another convention is that the package name is the base name of its source directory; the package in src/encoding/base64 is imported as "encoding/base64" but has name base64, not encoding_base64 and not encodingBase64.

https://go.dev/doc/effective_go#package-names

cc @0sewa0

Reply:
it package could be called `docker` and the file could be called `keychain.go`

docker is basically the same thing as `container registry` in our context, but shorter

Reply:
Correct, but also docker is not only a container registry it's also the name of the company, client, server, etc. :) 
I'm ok with your suggestion, at least it fits with the type inside keychain.go

Reply:
`containerregistry` because the package defines a component specifically to be used with the `go-containerregistry` library

Reply:
package name changed to more meaningful `dockerkeychain`